src/main/java/io/netty/buffer/api/ComponentProcessor.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api;
+
+import java.nio.ByteBuffer;
+
+/**
+ * This interface contain a collection of APIs used in the {@link Buf#forEachReadable(int, OfReadable)} and
+ * {@link Buf#forEachWritable(int, OfWritable)} methods.
+ */
+public interface ComponentProcessor {
+    /**
+     * A processor of {@linkplain ReadableComponent readable components}.
+     */
+    @FunctionalInterface
+    interface OfReadable extends ComponentProcessor {
```

Comment:
not a big fan of the name... Maybe its just because it starts with `Of...`..

Reply:
Same pattern as how `Spliterator` specialises.

Reply:
wasn't aware of this... so I guess it's fine. Matter of taste :)